cmd/podman/login.go

```diff
@@ -63,6 +67,28 @@ func login(cmd *cobra.Command, args []string) error {
 		skipTLS = types.NewOptionalBool(!loginOptions.tlsVerify)
 	}
 
+	secretName := cmd.Flag("secret").Value.String()
+	if len(secretName) > 0 {
+		if len(loginOptions.Password) > 0 {
+			return errors.New("--secret can not be used with --password options")
+		}
+		if len(loginOptions.Username) == 0 {
+			loginOptions.Username = secretName
+		}
+		var inspectOpts = entities.SecretInspectOptions{
+			ShowSecret: true,
+		}
+		inspected, errs, _ := registry.ContainerEngine().SecretInspect(context.Background(), []string{secretName}, inspectOpts)
+
+		if len(errs) > 0 && errs[0] != nil {
+			return errs[0]
+		}
+		if len(inspected) == 0 {
```

Comment:
Should we add a sanity check to avoid `len(inspected) > 1`?

Reply:
resolving this as @rhatdan added a new if below, and GitHub didn't see it.